### PR TITLE
Redesigned frontend using ant design with basic project and requisition setup

### PR DESCRIPTION
**Pull request by ayush-goyal**
Changes
----
Has pretty much the same features as the last pull request, like the user home page, project list, project detail page, requisition detail page, and requisition creation page, but recreated and redesigned with ant design. As of now, the functionality is just setup to pull from the backend, there are no mutations in place.

### CODE REVIEW

File: frontend/src/components/requisitionForm/RequisitionForm.tsx [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+          </Col>
+
+          <Col {...halfLayout}>
+            <Form.Item
+              name="paymentRequiredBy"
+              rules={[RULES.requiredRule]}
+              label={(
+                <span>
+                  {"Payment Required By "}
+                  <Tooltip title="The date you need the payment by (Should be at least 3 days from today)">
```
Comment (evan10s) [inline frontend/src/components/requisitionForm/RequisitionForm.tsx:143]: should probably put a TODO here to fill in the actual policy, but great that you're even thinking of this kind of stuff 💯

File: frontend/src/components/requisitionForm/RequisitionItemCard.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  <Card
+    size="small"
+    title={`Item ${props.field.name + 1}`}
+    style={{ marginBottom: "15px" }}
+    extra={props.deleteButton
+      ? (
+        <Button onClick={() => props.remove(props.field.name)} type="text" size="small" danger>
+          Delete
+        </Button>
+      ) : null}
```
Comment (evan10s) [inline frontend/src/components/requisitionForm/RequisitionItemCard.tsx:30]: You should be able shorten this to
```suggestion
    extra={props.deleteButton && (
        <Button onClick={() => props.remove(props.field.name)} type="text" size="small" danger>
          Delete
        </Button>
      )}
```

File: frontend/src/components/requisitionForm/RequisitionItemCard.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+          label={(
+            <span>
+              {"Link "}
+              <Tooltip title="The link to the item you want to purchase">
+                <QuestionCircleOutlined />
+              </Tooltip>
+            </span>
+          )}
+        >
+          <Input type="url" placeholder="http://rubberducks.com" />
```
Comment (evan10s) [inline frontend/src/components/requisitionForm/RequisitionItemCard.tsx:58]: this is an actual website that redirects to http://game-fundraising.com/ 😂

File: frontend/src/components/requisitionForm/RequisitionItemCard.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+          label="Quantity"
+        >
+          <InputNumber
+            prefix="$"
+            type="number"
+            min={1}
+            precision={0}
+            style={{ width: "100%" }}
+            placeholder="578"
+          />
```
Comment (evan10s) [inline frontend/src/components/requisitionForm/RequisitionItemCard.tsx:78]: - prefix shouldn't be $ if this is quantity
- precision of 0 seems to mean any decimal value entered gets rounded/truncated to 0 decimal places.  Even though this is a quantity input, I'm not sure this is entirely desirable because partial quantities are sometimes a thing

File: frontend/src/util/util.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    case "PENDING_CHANGES": return 1;
+    case "READY_TO_ORDER": return 2;
+    case "ORDERED": return 3;
+    case "RECEIVED": return 4;
+    case "CANCELLED": return 1;
+    default: return 0;
+  }
+};
+
+export const formatPrice = (num: number) => `$${num.toLocaleString("en-US", { minimumFractionDigits: 2, maximumFractionDigits: 2 })}`;
```
Comment (evan10s) [inline frontend/src/util/util.ts:44]: `Number.toLocaleString` seems to have a currency option you can use instead of the hard-coded dollar sign:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString

File: frontend/src/util/util.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+export const getTotalCost = (requisition: Requisition, includeOtherFees: boolean) => {
+  if (requisition && requisition.requisitionitemSet) {
+    const total = requisition.requisitionitemSet.map((item) => item.quantity * item.unitPrice).reduce((prev, curr) => prev + curr, 0);
```
Comment (evan10s) [inline frontend/src/util/util.ts:48]: `reduce` also iterates through the items; would it be possible to eliminate map and do this with just the one `reduce`?

File: frontend/src/App.tsx [hunk 7]
```diff
@@ -1,38 +1,50 @@
...
+import "./App.css";
+
+const { Header, Content, Footer } = Layout;
 
-        </div>
-    );
+function App() {
+  return (
+    <Router>
+      <Layout className="App" style={{ minHeight: "100vh" }}>
```
[the remark this answers is not in the document]
Comment (ayush-goyal) [reply frontend/src/App.tsx:19]: This is different from the hamburger component. This is to ensure the footer stays on the bottom of the page. Here are two screenshots, one with the minHeight, and one without the minHeight.
Screen Shot 2020-06-29 at 10 58 49 PM
Screen Shot 2020-06-29 at 10 59 17 PM

File: frontend/src/components/requisitionForm/RequisitionForm.tsx (hunk 1 above)
Comment (ayush-goyal) [reply frontend/src/components/requisitionForm/RequisitionForm.tsx:143]: Yeah, you're right Ill add it in

File: frontend/src/components/requisitionForm/RequisitionItemCard.tsx (hunk 4 above)
Comment (ayush-goyal) [reply frontend/src/components/requisitionForm/RequisitionItemCard.tsx:78]: I took out the dollar. Yes, that is the functionality. Currently, the db is setup to only accept positive integers. Should I go ahead and change everything to decimals?

File: frontend/src/components/requisitionForm/RequisitionItemCard.tsx (hunk 3 above)
Comment (ayush-goyal) [reply frontend/src/components/requisitionForm/RequisitionItemCard.tsx:58]: Haha that's funny

File: frontend/src/util/util.ts (hunk 5 above)
Comment (ayush-goyal) [reply frontend/src/util/util.ts:44]: Good find! Fixed.

File: frontend/src/components/requisitionForm/RequisitionItemCard.tsx (hunk 4 above)
Comment (evan10s) [reply frontend/src/components/requisitionForm/RequisitionItemCard.tsx:78]: Actually, I looked at our spring purchase requests and there were 0 with a decimal quantity.  So I think a positive integer is fine for that

File: frontend/src/App.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (evan10s) [reply frontend/src/App.tsx:19]: Makes sense
